lib/ansible/modules/storage/netapp/na_ontap_volume.py

```diff
@@ -0,0 +1,614 @@
+        if self.space_guarantee is not None:
+            options['space-reserve'] = self.space_guarantee
+        if self.volume_security_style is not None:
+            options['volume-security-style'] = self.volume_security_style
+
+        volume_create = netapp_utils.zapi\
+            .NaElement.create_node_with_children(
+                'volume-create', **options)
+        try:
+            self.server.invoke_successfully(volume_create,
+                                            enable_tunneling=True)
+            self.ems_log_event("create")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error provisioning volume %s \
+                                  of size %s: %s'
+                                  % (self.name, self.size, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def delete_volume(self):
+        '''Delete ONTAP volume'''
+        if self.is_infinite:
+            volume_delete = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-destroy-async', **{'volume-name': self.name})
+        else:
+            volume_delete = netapp_utils.zapi\
+                .NaElement.create_node_with_children(
+                    'volume-destroy', **{'name': self.name,
+                                         'unmount-and-offline': 'true'})
+        try:
+            self.server.invoke_successfully(volume_delete,
+                                            enable_tunneling=True)
+            self.ems_log_event("delete")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error deleting volume %s: %s'
+                                  % (self.name, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def move_volume(self):
+        '''Move volume from source aggregate to destination aggregate'''
+        volume_move = netapp_utils.zapi\
+            .NaElement.create_node_with_children(
+                'volume-move-start', **{'source-volume': self.name,
+                                        'vserver': self.vserver,
+                                        'dest-aggr': self.aggregate_name})
+        try:
+            self.cluster.invoke_successfully(volume_move,
+                                             enable_tunneling=True)
+            self.ems_log_event("move")
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error moving volume %s: %s'
+                                  % (self.name, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def rename_volume(self):
+        """
+        Rename the volume.
+
+        Note: 'is_infinite' needs to be set to True in order to rename an
+        Infinite Volume.
+        """
+        if self.is_infinite:
+            volume_rename = netapp_utils.zapi\
```

Comment:
Any way to block/poll for completion on this? Otherwise this is a built-in race condition for anyone that wants to do anything with a renamed infinite volume from Ansible...

Reply:
We will look at this in the next release.